src/Contract/Operation/Appendable.php

```diff
@@ -21,9 +21,11 @@ interface Appendable
     /**
      * Add one or more items to a collection.
      *
-     * @param mixed ...$items
+     * @template A
      *
-     * @psalm-return \loophp\collection\Collection<int|TKey, T>
+     * @param A ...$items
+     *
+     * @return \loophp\collection\Collection<int|TKey, A|T>
```

Comment:
@drupol this is out of sync with the [Append operation](https://github.com/loophp/collection/blob/master/src/Operation/Append.php#L24), which I also want to update.

First I wanted to understand: is `append` meant to allow adding items of any type to the collection? Or only items of the type already contained? Example below:

```PHP

A) Collection::fromIterable([1, 2, 3])->append(4); // this is always fine
B) Collection::fromIterable([1, 2, 3])->append('bar'); // should this be allowed?
```

B) transforms the collection from `Collection<int, int>` into `Collection<int, string|int>`. I know this library supports this behaviour, but I'm not sure what your intention was when building it.

Reply:
Yes, mixing types is definitely allowed.

Reply:
I see, in that case I think static analysis becomes less useful in many cases.
For `append` if it can be used like this `->append(4, 'foo', ['bar'])` then I guess `mixed` is the only type that can be used for the `$items`.

The problem is then that even if you do `A)` (append the same type) the static analysis will not be able to tell that the collection is still all of the same type.

Reply:
I have mixed (pun intended 😄 ) feelings about this. On one hand I see the flexibility of mixing types but it's a bit inconvenient when the collection is used as a single-type collection because after many operations type hints will need to be used in the calling code to show that the type is still the same

Reply:
I sharing the feelings.

However, this library has never been made to be a single-type collection.

I know that this would make typing more complex or even impossible, but this is how it is. I think we should keep that flexibility for the ease of use.

If we really want a strict collection, maybe we should use a `StrictCollection` class and do the stuff in it.

WDYT?

Reply:
I would definitely like an option for a strictly-typed collection 👍 . Maybe we could call it `TypedCollection`.

How would it work with the type hints fo the operations though? Would the code effectively need to be duplicated but with different type hints?

Reply:
Good question. I think this is part of a much broader discussion in its own branch, don't you think?

Reply:
yep I agree, I might start one in https://github.com/loophp/collection/discussions at some point?

For this PR I'll update things to take into account collection usage with data of different types

Reply:
A discussion is fine to start with, then a PR is also fine. There will be so much things to discuss that it's better to dedicate its own scope.